### PR TITLE
feat(AddDataset): AddDataset can now take a dataset .yaml or .json file as well as a body file

### PR DESCRIPTION
**Pull request by ramfox**
With the CLI, you can add a dataset using a dataset .yaml or .json file as well as from a body.csv file.

On the frontend, you could only add from a body.csv/.json file.

We have higher ambitions for the AddDataset component, but for now, you can now upload both a body and dataset file when you are adding a dataset to qri.

### CODE REVIEW

File: lib/scss/mixins/_forms.scss [hunk 1]
```diff
@@ -83,3 +83,28 @@
     height: auto;
   }
 }
+
+// Radio variant
+// 
+// Used in _forms to create the different colors that the radio button can have
+@mixin radio-variant($checked-color, $background-color) {
```
Comment (b5) [inline lib/scss/mixins/_forms.scss:90]: 👌

File: lib/actions/dataset.js [hunk 2]
```diff
@@ -242,7 +242,7 @@ export function downloadDataset (datasetRef) {
   }
 }
 
-export function initDataset (name, files, url) {
+export function initDataset (name, file, body, url) {
```
Comment (b5) [inline lib/actions/dataset.js:245]: as a general note, "file" is a little ambiguous here. I think we should work on a better convention for referring to dataset document files. _but_ this is pulled from the CLI, so let's not worry about it now.

File: lib/components/AddDataset.js [hunk 3]
```diff
@@ -24,9 +25,11 @@ export default class AddDataset extends Base {
       dataset: {
         name: '',
         url: '',
-        files: undefined,
-        file_input: undefined
-      }
+        file: undefined,
+        body: undefined,
+        body_input: undefined
```
Comment (b5) [inline lib/components/AddDataset.js]: any reason to not use `bodyInput`?

File: lib/components/form/RadioInput.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    const {name, value, text, color, defaultChecked, onChange} = this.props
+    const className = 'radio-' + color
+    return (
+      <span className={css('span')}><input className={className} type='radio' name={name} value={value} defaultChecked={defaultChecked} onChange={onChange} />{text}</span>
+    )
+  }
+  styles () {
+    return {
+      span: {
+        // display: 'flex'
```
Comment (b5) [inline lib/components/form/RadioInput.js]: ded code. can remove whole span definition by the looks of it

**Review by b5 — COMMENTED** [review]
Looks good, only a few nitpiks

File: lib/actions/dataset.js (hunk 2 above)
Comment (ramfox) [reply lib/actions/dataset.js:245]: I think about this every time I look at the CLI, my gut reaction is to change it to `dataset`. Following an html convention we can maybe call it `headers` but I'm not sure I'm feeling that.

**Comment by b5** [comment]
🌟 🎷 🐩

**Pull request merged**